### PR TITLE
#1688: When resolve assembly from GAC compare version

### PR DESCRIPTION
**Pull request by troydai**
Fix issue #1688 

I'm adding tests right now.

### CODE REVIEW

**Comment by troydai** [comment]
Please review: @davidfowl @anurse @BrennanConroy @victorhurdugaci

**Comment by BrennanConroy** [comment]
Looks good, :watch: tests

**Comment by troydai** [comment]
Ran the tests locally on Mac, they're indeed skipped.

**Comment by BrennanConroy** [comment]
OSSkipCondition should probably use "Linux" instead of "Unix"

**Comment by troydai** [comment]
No such option: https://github.com/aspnet/Testing/blob/dev/src/Microsoft.AspNet.Testing/xunit/OperatingSystems.cs

**Comment by BrennanConroy** [comment]
Ya I know, I'm talking about the internals of the class

File: test/Microsoft.Framework.Runtime.Tests/GacDependencyResolverFacts.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+using Microsoft.Framework.Runtime.Helpers;
+using NuGet;
+using Xunit;
+
+namespace Microsoft.Framework.Runtime.Tests
+{
+    public class GacDependencyResolverFacts
+    {
+        [ConditionalTheory]
+        [OSSkipCondition(OperatingSystems.MacOSX | OperatingSystems.Linux)]
```
Comment (davidfowl) [inline test/Microsoft.Framework.Runtime.Tests/GacDependencyResolverFacts.cs]: skip mono as well.

**Comment by troydai** [comment]
Ping @davidfowl  Sign off?

File: test/Microsoft.Framework.Runtime.Tests/GacDependencyResolverFacts.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using Microsoft.Framework.Runtime.Helpers;
+using NuGet;
+using Xunit;
+
+namespace Microsoft.Framework.Runtime.Tests
+{
+    public class GacDependencyResolverFacts
+    {
+        [ConditionalTheory]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono)]
```
Comment (davidfowl) [inline test/Microsoft.Framework.Runtime.Tests/GacDependencyResolverFacts.cs]: Why did you remove the other conditional?

**Comment by davidfowl** [comment]
Re reun the build

**Comment by davidfowl** [comment]
:shipit:

**Comment by troydai** [comment]
I'll merge this change as soon as the CI passes.
